### DISCUSSION TITLE
Keep track of resources that Cloud Custodian removes due to violations?

### DISCUSSION THREAD

**Discussion by haiauth** [cloud-custodian/cloud-custodian#7453] [General]
### Ask your question

Is there any way that I can keep a record of or keep track of what resources Cloud Custodian removes due to violations? 

Like the example policy below, how can I know which access key, where it lives, and what its name and when it was removed by Cloud Custodian because the key is too old and active so it needed to be rotated? 

UPDATED: 

Looks like there is a flag called `--metrics aws` that I can used, but does `c7n-org` support this feature? If so, how do I use it if I want to forward all the metrics to ONE account which is the one that I am using to run Cloud Custodian from?

### Policy

```shell
policies: 
  - name: iam-user-old-active-access-key
    resource: iam-user
    comment: |
      Checks if either of a user's active keys are older than 180 days. 
    filters:
      - type: access-key
        key: Status
        value: Active
      - or:
        - type: value
          key: '"c7n:AccessKeys"[?Status == ''Active''].CreateDate | [0]'
          value: 180
          op: greater-than
          value_type: age
        - type: value
          key: '"c7n:AccessKeys"[?Status == ''Active''].CreateDate | [1]'
          value: 180
          op: greater-than
          value_type: age
    actions: 
      - type: notify
        slack_template: slack-iam-old-active-access-key
        slack_msg_color: danger
        violation_desc: |
          One or more user(s) who has an ACTIVE access key which has not been used for the last 180 days.
        action_desc: |
          Consider remove the access key(s) and do a key rotation.
        to:
          - webhook
        transport:
          type: sqs
          queue: https://sqs.us-east-2.amazonaws.com/<acct_id>/<sqs_queue_name>
```

### Relevant log/traceback output

```shell
n/a
```

**Comment by jtroberts83** [comment]
@haiauth There are a couple ways to accomplish this with Cloud Custodian.

First option is to create a Cloud Custodian mailbox and then have all policies that take an action on a resource or delete a resource send a notification to your Cloud Custodian mailbox so you have an audit history via emails on all the actions custodian has taken then it's just a matter of searching the mailbox for the resource name or account etc.

The second option would be to output all your Custodian logs to a versioned S3 bucket and then you could have a lambda function that runs on PutObjects in that bucket and it can analyze and pull out important events that you care about into a report or you can run a tool of your choice to ingest the S3 data and analyze it.
